test/unit/features/instance/methods-lifecycle.spec.js

```diff
@@ -53,6 +53,37 @@ describe('Instance methods lifecycle', () => {
         }
       }).$mount()
     })
+
+    it('Dep.target should be undefined during invocation of child immediate watcher', done => {
+      let calls = 0, childData
+      const parentUpdate = jasmine.createSpy()
+      new Vue({
+        template: '<div><my-component></my-component></div>',
+        updated: parentUpdate,
+        components: {
+          myComponent: {
+            template: '<div>{{ a }}</div>',
+            data() {
+              return childData = { a: 123 }
+            },
+            watch: {
+              a: {
+                handler() {
```

Comment:
let's use a spy here too since the Dep.target is an implementation detail

Reply:
Are you suggesting not having our own function there? When the watcher callback is called it either needs to check Dep.target or touch childData.a to cause the issue. Should I just replace the Dep.target check with a `let x = this.a`, and then we rely on the `expect(parentUpdate).not.toHaveBeenCalled()`?

Reply:
I'm realizing I was giving you hints as if they were jest spies but they are not. The `toHaveBeenCalled()` seems to work though. There is probably a way to mock the implementation of the spy but it's not necessary, it might be an outdated version of Jasmine.
Reading `this.a` is better, yes